src/Factory/ProductDocumentFactory.php

```diff
@@ -72,40 +86,57 @@ public function __construct(
+        $this->priceDocumentClass = $priceDocumentClass;
+
+        $this->assertClassExtends($variantDocumentClass, VariantDocument::class);
+        $this->variantDocumentClass = $variantDocumentClass;
+
+        $this->assertClassExtends($optionDocumentClass, OptionDocument::class);
+        $this->optionDocumentClass = $optionDocumentClass;
 
         $this->attributeWhitelist = $attributeWhitelist;
     }
 
     /**
      * @param ProductInterface $syliusProduct
-     * @param LocaleInterface $locale
+     * @param LocaleInterface  $locale
      * @param ChannelInterface $channel
      *
      * @return ProductDocument
      */
-    public function createFromSyliusSimpleProductModel(ProductInterface $syliusProduct, LocaleInterface $locale, ChannelInterface $channel)
-    {
-        if (!$syliusProduct->isSimple()) {
-            throw new UnsupportedFactoryMethodException(
-                __METHOD__,
-                sprintf(
-                    'Cannot create elastic search model from configurable product "%s".',
-                    $syliusProduct->getCode()
-                )
-            );
-        }
+    public function createFromSyliusSimpleProductModel(
+        ProductInterface $syliusProduct,
+        LocaleInterface $locale,
+        ChannelInterface $channel
+    ): ProductDocument {
 
-        /** @var ProductVariantInterface $productVariant */
-        $productVariant = $syliusProduct->getVariants()->first();
+        $minProductChannelPrice = $this->getChannelPricingForChannelFromProductVariant(
+            $syliusProduct->getVariants()->first(),
+            $channel
+        );
+        /** @var ProductVariantInterface[] $syliusProductVariants */
+        $syliusProductVariants = $syliusProduct->getVariants();
+
+        /**
+         * Select minimal product price out of all variants
+         */
+        $variants = [];
+        foreach ($syliusProductVariants as $variant) {
+            $channelPrice = $this->getChannelPricingForChannelFromProductVariant($variant, $channel);
+            if ($minProductChannelPrice->getPrice() < $channelPrice->getPrice()) {
+                $minProductChannelPrice = $channelPrice;
+            }
+            $variants[] = $this->createVariantDocumentFromSyliusVariant($variant, $channel);
+        }
 
         /** @var ProductTranslationInterface|TranslationInterface $productTranslation */
-        $productTranslation = $syliusProduct->getTranslation($locale->getCode());
-        $channelPrice = $this->getChannelPricingForChannelFromProductVariant($productVariant, $channel);
+        $productTranslation      = $syliusProduct->getTranslation($locale->getCode());
         $syliusProductAttributes = $syliusProduct->getAttributesByLocale(
             $locale->getCode(),
             $channel->getDefaultLocale()->getCode()
         );
-        $syliusProductTaxons = $syliusProduct->getProductTaxons();
+        $syliusProductTaxons     = $syliusProduct->getProductTaxons();
```

Comment:
Heh, code auto formatting, i'll change the rules, but I did like how it looked though :)